pageserver/src/walredo.rs

```diff
@@ -554,6 +555,28 @@ impl PostgresRedoManager {
     }
 }
 
+///
+/// Command with ability not to give all file descriptors to child process
+///
+trait CloseFileDescriptors: CommandExt {
+    ///
+    /// Close file descriptors (other than stdin, stdout, stderr) in child process
+    ///
+    fn close_fds(&mut self) -> &mut Command;
+}
+
+impl<C: CommandExt> CloseFileDescriptors for C {
+    fn close_fds(&mut self) -> &mut Command {
+        unsafe {
+            self.pre_exec(move || {
+                // close_fds crate is safe to use in this context (see docs)
```

Comment:
NIT: Simple folk like me would really thank you for writing a SAFETY paragraph with explicit explanation, why it's safe.
One of the many examples: https://github.com/rust-lang/rust/blob/a28109a7679fbb254ca0962865a8b32b91ddbbd9/library/std/src/sys/sgx/waitqueue/unsafe_list.rs#L41

That would also help to validate the assumptions: "see docs" would require a person to construct his own and guess for you instead, that does not help to validate things simple.